### PR TITLE
Jim Dechant 1710 BE

### PR DESCRIPTION
**Pull request by Ecksi**
Scrabble

### CODE REVIEW

File: test/scrabble_test.rb [hunk 1]
```diff
@@ -6,8 +6,26 @@
...
+    assert_equal 18, Scrabble.new.score("Do not give up")
+  end
+
+  def test_it_can_score_an_empty_string
+    assert_equal 0, Scrabble.new.score("")
+    assert_equal 0, Scrabble.new.score(" ")
+  end
+
+  def test_can_it_use_word_multipliers
+    hacked together...
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb]: Make sure to remove text that will error your work before pushing.

File: test/scrabble_test.rb [hunk 2]
```diff
@@ -6,8 +6,26 @@
...
+  end
+
+  def test_it_can_score_a_multi_word_string
+    assert_equal 16, Scrabble.new.score("I can do this")
+    assert_equal 18, Scrabble.new.score("Do not give up")
+  end
+
+  def test_it_can_score_an_empty_string
+    assert_equal 0, Scrabble.new.score("")
+    assert_equal 0, Scrabble.new.score(" ")
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb]: Way to consider extra edge cases.

File: test/scrabble_test.rb [hunk 3]
```diff
@@ -6,8 +6,26 @@
...
   end
+
+  def test_it_can_score_a_string
+    assert_equal 9, Scrabble.new.score("help")
+    assert_equal 10, Scrabble.new.score("smush")
+  end
+
+  def test_it_can_score_a_multi_word_string
+    assert_equal 16, Scrabble.new.score("I can do this")
+    assert_equal 18, Scrabble.new.score("Do not give up")
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb]: The spec (nor the game scrabble) never asks for functionality with multiple words. Make sure you're reading the spec thoroughly before developing added functionality.

File: lib/scrabble.rb [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
+    final_result = multipler_array[0..remove_excess].sum
+    if word_multiplier == 1
+      final_result
+    # elsif word.length >= 7
+    #   final_result = (final_result * word_multiplier) + 10
+    else
+      final_result = final_result * word_multiplier
+    end
+    final_result
   end
```
Comment (VictoriaVasys) [inline lib/scrabble.rb]: You identified that this method is very long; it would be helpful for your learning if you refactored (make sure you have robust tests before changing too much). Also, consider various enumerables you could reach for instead of using a counter.

File: lib/score.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    if word == nil
+      score_tally << point_values[nil]
+    else
+      convert_word.each do |convert|
+        score_tally << point_values[convert]
+      end
+    end
+    score_tally
+  end
```
Comment (VictoriaVasys) [inline lib/score.rb]: Using map here would be more appropriate (and help decrease the length of this method).

File: lib/scrabble.rb [hunk 6]
```diff
@@ -1,18 +1,33 @@
...
+      end
+    end
+    count += 1
+
+    remove_excess = word.length - 1
+    final_result = multipler_array[0..remove_excess].sum
+    if word_multiplier == 1
+      final_result
+    # elsif word.length >= 7
+    #   final_result = (final_result * word_multiplier) + 10
```
Comment (VictoriaVasys) [inline lib/scrabble.rb]: Remove comments before pushing your submission.

File: lib/scrabble.rb [hunk 7]
```diff
@@ -1,18 +1,33 @@
...
-      "Y"=>4, "Z"=>10
-    }
+  def score_with_multipliers(word, letter_multiplier, word_multiplier = 1)
+    score_and_multiplier = score_finder(word).zip(letter_multiplier)
+    multipler_array = []
+    count = 0
+
+    score_and_multiplier.each do |main_array|
+      score_and_multiplier.each do |sub_set|
+        multipler_array << sub_set[count] * sub_set[count +1]
```
Comment (VictoriaVasys) [inline lib/scrabble.rb]: Your `rake` command is giving extra clues as to how to improve your code. Try looking up the problem regarding unary v. binary operators here.

File: lib/scrabble.rb [hunk 8]
```diff
@@ -1,18 +1,33 @@
...
-      "E"=>1, "F"=>4, "G"=>2, "H"=>4,
-      "I"=>1, "J"=>8, "K"=>5, "L"=>1,
-      "M"=>3, "N"=>1, "O"=>1, "P"=>3,
-      "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
-      "U"=>1, "V"=>4, "W"=>4, "X"=>8,
-      "Y"=>4, "Z"=>10
-    }
+  def score_with_multipliers(word, letter_multiplier, word_multiplier = 1)
+    score_and_multiplier = score_finder(word).zip(letter_multiplier)
+    multipler_array = []
```
Comment (VictoriaVasys) [inline lib/scrabble.rb]: Shy away from using data types in naming (don't use 'array' or 'string', etc. in your ruby names)

File: lib/score.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
+module ScoreGenerator
+  def score_finder(word)
+    convert_word = word.upcase.chars
+    score_tally = []
+
+    if word == nil
+      score_tally << point_values[nil]
+    else
+      convert_word.each do |convert|
```
Comment (VictoriaVasys) [inline lib/score.rb]: Think about more appropriate naming for your block parameter (what does it actually represent?).

File: lib/score.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
+module ScoreGenerator
```
Comment (VictoriaVasys) [inline lib/score.rb]: I like that you're exploring modules & there might be a use-case for pulling out the "dictionary" of score letters, but score generator would probably also encompass all of your other scoring functionality (which is basically the whole scrabble class).
